lib/reader/src/parser.rs

```diff
@@ -65,6 +86,12 @@ pub struct Parser<'a> {
 
     // Comments collected so far.
     comments: Vec<Comment<'a>>,
+
+    // Reference to the target_isa for things like parsing ISA-specific instruction encoding
+    // information. This is only `Some` if exactly one set of `isa` directives were found in the
+    // prologue (it is valid to have directives for multiple different ISAs, but in that case we
+    // couldn't know which ISA the provided encodings are intended for)
+    target_isa: Option<Box<TargetIsa>>,
```

Comment:
Here. Use this instead:

```rust
unique_isa: Option<&TargetIsa>,
```

(I prefer the name `unique_isa` so people won't be surprised when it is missing.)